agent/rpc/peering/service.go

```diff
@@ -427,6 +431,19 @@ type BidirectionalStream interface {
 
 // StreamResources handles incoming streaming connections.
 func (s *Service) StreamResources(stream pbpeering.PeeringService_StreamResourcesServer) error {
+	if s.Backend == nil {
+		s.logger.Error("cannot establish stream without a backend", "error", errPeeringBackendNil)
+		return errPeeringBackendNil
+	}
+
+	if !s.Backend.IsLeader() {
+		// we are not the leader so we will hang up on the dialer
+
+		// TODO(peering): in the future we want to indicate the address of the leader server as a message to the dialer (best effort, non blocking)
+		s.logger.Error("cannot establish a peering stream on a follower node;")
+		return io.EOF
```

Comment:
I think you also need to do something similar down around line 581 in the `for...select` loop in HandleResources.

Reply:
thanks for all the guidance last week; added a similar check in `HandleStream`